examples/embeddable_explorer/public/todo_embeddable_example.tsx

```diff
@@ -19,86 +19,55 @@
 
 import React from 'react';
 import {
-  EuiFormRow,
+  EuiButton,
   EuiFieldText,
-  EuiPanel,
-  EuiTextArea,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
   EuiPageBody,
   EuiPageContent,
   EuiPageContentBody,
   EuiPageHeader,
   EuiPageHeaderSection,
-  EuiTitle,
+  EuiPanel,
   EuiText,
-  EuiButton,
-  EuiFlexGroup,
-  EuiFlexItem,
+  EuiTextArea,
+  EuiTitle,
 } from '@elastic/eui';
-import {
-  TodoEmbeddable,
-  TODO_EMBEDDABLE,
-  TodoInput,
-} from '../../../examples/embeddable_examples/public/todo';
-import {
-  EmbeddableStart,
-  EmbeddableRoot,
-  EmbeddableOutput,
-  ErrorEmbeddable,
-} from '../../../src/plugins/embeddable/public';
+import { TodoInput } from '../../../examples/embeddable_examples/public/todo';
+import { EmbeddableRenderer } from '../../../src/plugins/embeddable/public';
+import { EmbeddableExamplesStart } from '../../embeddable_examples/public/plugin';
 
 interface Props {
-  getEmbeddableFactory: EmbeddableStart['getEmbeddableFactory'];
+  todoEmbeddableFactory: EmbeddableExamplesStart['todoEmbeddableFactory'];
 }
 
 interface State {
   task?: string;
   title?: string;
   icon?: string;
   loading: boolean;
+  input: TodoInput;
 }
 
 export class TodoEmbeddableExample extends React.Component<Props, State> {
-  private embeddable?: TodoEmbeddable | ErrorEmbeddable;
-
   constructor(props: Props) {
     super(props);
 
-    this.state = { loading: true };
-  }
-
-  public componentDidMount() {
-    const factory = this.props.getEmbeddableFactory<TodoInput, EmbeddableOutput, TodoEmbeddable>(
-      TODO_EMBEDDABLE
-    );
-
-    if (factory === undefined) {
-      throw new Error('Embeddable factory is undefined!');
-    }
-
-    factory
-      .create({
+    this.state = {
+      loading: true,
+      input: {
         id: '1',
         task: 'Take out the trash',
         icon: 'broom',
         title: 'Trash',
-      })
-      .then((embeddable) => {
-        this.embeddable = embeddable;
-        this.setState({ loading: false });
-      });
-  }
-
-  public componentWillUnmount() {
-    if (this.embeddable) {
-      this.embeddable.destroy();
-    }
+      },
+    };
   }
 
   private onUpdateEmbeddableInput = () => {
-    if (this.embeddable) {
-      const { task, title, icon } = this.state;
-      this.embeddable.updateInput({ task, title, icon });
```

Comment:
Not sure why this was a preferred way? 🤔 
new `EmbeddableRenderer` supports passing input as props. 
What do you think about it?

Reply:
Nice!  I like it better this way!